component/src/ibc/ibc_token.rs

```diff
@@ -0,0 +1,57 @@
+use ibc::core::ics24_host::identifier::{ChannelId, PortId};
+use penumbra_crypto::asset;
+use sha2::{Digest, Sha256};
+
+/// IBC token respresents a token that was created through IBC.
+pub struct IBCToken {
+    transfer_path: String,
+    base_denom: asset::Denom,
+}
+
+/// <https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-001-coin-source-tracing.md>
+fn derive_ibc_denom(transfer_path: &str) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(transfer_path.as_bytes());
+
+    let denom_bytes = hasher.finalize();
+    let denom_hex = String::from_utf8(hex::encode_upper(denom_bytes).into()).unwrap();
+
+    format!("ibc/{}", denom_hex)
```

Comment:
Do we want to use the `ibc/HASH` as the base denom, or the full transfer path?

Our asset IDs are already hashes of the full transfer path, so I'm wondering if our version of the "ibc denom" is already our asset ID.

If we store the transfer path as the denom, we can also have methods on the IBC denom that would do things like parse it to extract the chain of bearer assets, etc.

Reply:
This design follows the "coin source tracing" ADR as proposed in ibc-go:

https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-001-coin-source-tracing.md

afaik, the two main problems that caused this designed to be reached were SDK-specific issues: without hashing, the denom can grow indefinitely as transfers are made across zones, which is a problem in the cosmos SDK due to the Coin type only supporting denoms of up to 64 character len, and the SDK Coin type applies some regex to the denom. 

Given that our asset IDs already hash the full transfer path, I think it makes sense to just have our IBC denom be the asset ID.

Reply:
ah, come to think of it, i think this approach would cause issues for our counterparties (SDK chains), for the reasons noted in that ADR: if we were to have our IBC denom be the penumbra asset ID (e.g., `ibc/passet194...`), then the denoms would be too large for the SDK.Coin type.